src/main/java/org/spongepowered/common/entity/EntityUtil.java

```diff
@@ -317,6 +317,8 @@ public static boolean isEntityDead(net.minecraft.entity.Entity entity) {
         final IPhaseState state = peek.state;
         final PhaseContext<?> context = peek.context;
 
+        Sponge.getCauseStackManager().pushCause(entityIn);
```

Comment:
It doesn't seem like you're popping the cause after pushing. You need to ensure that the cause stack is stack neutral after the method is completed, so either popping the cause, or wrapping with a try with resources on a stack frame.

Reply:
Where and with what should I pop the cause ?
I don't know if I should really pop the cause since the method is called inside a try-with-resources (in fact, I'm not event sure I should do [that](https://github.com/SpongePowered/SpongeCommon/pull/1741/files/5d9823590665615f835869725b3aa037036e7478#diff-8455601503527079e2c6f540298a089eR62))